### ISSUE TITLE
Initial zone set-up does not verify usable secondary storage - leads to issues

### ISSUE THREAD

**Issue by Jayd603** [apache/cloudstack#8995]
Latest cloudstack, 
I just did a new cloudstack install, set up first zone, I accidentally used the wrong secondary NFS storage path, the set-up process gave no errors. Now, after creating new and correct secondary storage, I cannot migrate and cannot delete the old one. Zone set-up script should test connection and write ability of the secondary storage I'm thinking. Now i'm just looking for best way to fix, might just be easier to destroy the zone and re-create.

Oh, and, there are now templates and ISOs in "Not Ready" state with no option to remove them in the UI.

I'm trying to fix in the db now.

**Comment by DaanHoogland** [comment]
@Jayd603 , is your system vm template seeded on your secondary storage? The path `/template/tmpl/1/3` on your secondary storage should contain a valid systemvm template, something that should have happened automagically when the installation process would have gone flawless.

**Comment by Jayd603** [comment]
> @Jayd603 , is your system vm template seeded on your secondary storage? The path `/template/tmpl/1/3` on your secondary storage should contain a valid systemvm template, something that should have happened automagically when the installation process would have gone flawless.

Yeah, the NFS client says "protocol not supported" when it can't find a directory, just a tiny bit misleading response there. : - )   ... so, what actually happened was on initial zone set-up, the set up utility let me configure a non-usable NFS secondary server and carried on with no errors, this required editing the db and doing some digging to fix.  It's a sort of bug, since it should not allow a non-working secondary storage server to be added.

**Comment by Jayd603** [comment]
> @Jayd603 , can you remove the zone/storage and re-add instead?

It takes a lot of steps to delete a zone and I don't remember if it removed the stale/bad secondary storage entry.  I think I added a new one ok and everything worked but the old one still needed to be removed in the DB.  I /might/ be wrong about that, but there should still be some type of basic check when adding secondary storage to prevent issues.   --- it's obviously a minor issue, I have this cluster up and running , just need to add more host machines now.  thanks
